src/generate_charts.py

```diff
@@ -153,6 +153,19 @@ def generate_gantt_chart(
         category_orders={COLOR_LABEL: color_labels},
     )
 
+    # Update legend to remove conflict text and duplicates.
+    unique_labels = set()
+    for i, trace in enumerate(fig.select_traces()):
+        label = trace.legendgroup.split(",")[0]
+        if label not in unique_labels:
+            trace.update({"name": label, "legendrank": i})
+            unique_labels.add(label)
+        else:
+            fig.update_traces({"showlegend": False}, {"name": label})
+            trace.update({"name": label, "legendrank": i})
```

Comment:
You could also just hide the second instance of each occurrence, unless you deliberately want those to not have the pattern in the legend, which I'm not sure makes more sense.
```suggestion
            trace.update({"showlegend": False})
```

Reply:
Yeah I don't really have a preference either way, I'm fine with this way